tools/rpkg/src/scan.cpp

```diff
@@ -6,20 +6,262 @@
+		case PhysicalType::UINT8:
+			AppendColumnSegment<int, uint8_t, RFactorType>(data_ptr, sexp_offset, v, this_count);
+			break;
+
+		case PhysicalType::UINT16:
+			AppendColumnSegment<int, uint16_t, RFactorType>(data_ptr, sexp_offset, v, this_count);
+			break;
+
+		case PhysicalType::UINT32:
+			AppendColumnSegment<int, uint32_t, RFactorType>(data_ptr, sexp_offset, v, this_count);
+			break;
+
+		default:
+			cpp11::stop("rapi_execute: Unknown enum type for scan: %s",
+			            TypeIdToString(v.GetType().InternalType()).c_str());
+		}
+		break;
+	}
+	case RType::TIMESTAMP: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, timestamp_t, RTimestampType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_SECONDS: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, dtime_t, RTimeSecondsType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_MINUTES: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, dtime_t, RTimeMinutesType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_HOURS: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, dtime_t, RTimeHoursType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_DAYS: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, dtime_t, RTimeDaysType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_WEEKS: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, dtime_t, RTimeWeeksType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_SECONDS_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, dtime_t, RTimeSecondsType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_MINUTES_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, dtime_t, RTimeMinutesType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_HOURS_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, dtime_t, RTimeHoursType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_DAYS_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, dtime_t, RTimeDaysType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::TIME_WEEKS_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, dtime_t, RTimeWeeksType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::DATE: {
+		auto data_ptr = (double *)coldata_ptr;
+		AppendColumnSegment<double, date_t, RDateType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::DATE_INTEGER: {
+		auto data_ptr = (int *)coldata_ptr;
+		AppendColumnSegment<int, date_t, RDateType>(data_ptr, sexp_offset, v, this_count);
+		break;
+	}
+	case RType::LIST_OF_NULLS:
+	case RType::BLOB: {
+		auto data_ptr = (SEXP *)coldata_ptr;
```

Comment:
`BLOB` and `LIST_OF_NULLS` had different logic here https://github.com/duckdb/duckdb/pull/8503/files#diff-37ff3203dc9d0866d479ae910101108e08a2bdcf1f93bcf3f357c3be6adc94b3L371 . 
Why are they merged together now?